### ISSUE TITLE
When trying to use the library, CORS policy error arrises

### ISSUE THREAD

**Issue by jisraeltoledo** [boxing/boxrec#151]
Access to fetch at 'http://boxrec.com/en/boxer/659461' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by mikedidomizio** [comment]
Hey @jisraeltoledo 

It should be run server side NodeJS.
If you're accessing/running this in a browser, you'll run into CORS issues

**Comment by jisraeltoledo** [comment]
You were right. In the server side it doesn't sends the CORS error, however:

Error: Could not get response from boxrec
    at Function.<anonymous> (/srv/node_modules/boxrec-requests/dist/boxrec-requests.js:484:31)
    at step (/srv/node_modules/boxrec-requests/dist/boxrec-requests.js:32:23)
    at Object.throw (/srv/node_modules/boxrec-requests/dist/boxrec-requests.js:13:53)
    at rejected (/srv/node_modules/boxrec-requests/dist/boxrec-requests.js:5:65)
    at <anonymous>

That I received with the sample code you got.

exports.fetchBoxRec = functions.https.onRequest(async (req, res) => {
  let boxerId = req.query.boxerId;
  console.log("boxerId", boxerId);

  try {
    const cookieJar = await boxrec.login("", ""); // valid credentials to login
    const gennadyGolovkin = await boxrec.getPersonById(cookieJar, boxerId);
    console.log(gennadyGolovkin.name); // Gennady Golovkin
    console.log(gennadyGolovkin.division); // middleweight
    console.log(gennadyGolovkin.titlesHeld); // currently held titles
    console.log(gennadyGolovkin.otherInfo); // other info that couldn't be categorized
    console.log(gennadyGolovkin.suspended); // will tell if boxer is currently suspended
    console.log(gennadyGolovkin.bouts); // list of bouts
    console.log(gennadyGolovkin.bouts[37].opponent.name); // Saul Alvarez

    res.status(200).send(JSON.stringify(gennadyGolovkin));
    return true;
  } catch (e) {
    // error occurred logging in
    console.log(e);
    res.status(500).send(JSON.stringify(e));
    return false;
  }
});

Do you know what am I doing wrong?

**Comment by mikedidomizio** [comment]
So `Could not get response from boxrec` comes from https://github.com/boxing/boxrec-requests/blob/master/src/boxrec-requests.ts#L356 when attempting to create/get a session on BoxRec.

- Can you `ping boxrec.com` from this machine?
- Could you install `request-promise` temporarily and make a request like:

```
const rp = require('request-promise');
....
async function() {
   const options = {
            resolveWithFullResponse: true,
            uri: "http://boxrec.com",
        };

        await rp.get(options).then((data) => data.headers["set-cookie"]); // <--- What is this??
}
```
You'll have to change the above a bit to work with what you have

I'm curious what  `data.headers["set-cookie"]` is, if the cookie isn't being returned you'll get your error.

**Comment by jisraeltoledo** [comment]
This is what returns in (data.headers["set-cookie"])

["PHPSESSID=9fbndjvataqtd1n9kod9roh5fd; path=/; HttpOnly"]

I'm using firebase functions, and it doesn't send the ping correctly, I tried to send a ping with https://www.npmjs.com/package/ping but it didn't work.

**Comment by mikedidomizio** [comment]
So that response looks fine, that's what you want.  It seems like there's possibly another issue that is giving you `Could not get response from boxrec`.  I'm not entirely sure at this point what is the issue.

You're welcome, glad to help.  I wish I could help out more with this issue

**Comment by mikedidomizio** [comment]
Hey @jisraeltoledo,

I'm closing this issue since it's been a month since the last comment.  Hopefully you resolved your issue and if you think it'll be any help to others feel free to comment below.

Thanks
